src/configmanager.cpp

```diff
@@ -3546,3 +3554,36 @@ void ConfigManager::updateManagedOptionObjects(ManagedProperty *property)
 	foreach (QObject *o, managedOptionObjects[property].second)
 		property->writeToObject(o);
 }
+
+#ifdef Q_OS_WIN
+void ConfigManager::searchLaTeX()
+{
+	QString fileName = QFileDialog::getOpenFileName(nullptr, tr("Browse program"), "", tr("pdflatex.exe"));
+	if (fileName.isEmpty()) return;
+	QFileInfo fi(fileName);
+	if (!fi.exists()) return;
+	QString tmpMiktex = QDir::toNativeSeparators(fi.absoluteDir().absolutePath()) + fi.absoluteDir().separator();
+	QString tmpTexlive = tmpMiktex;
+	buildManager->swapWinSearchPath(tmpMiktex, tmpTexlive);
+	QPushButton *searchButton = qobject_cast<QPushButton *>(sender());
+	QWidget *parentWidget = searchButton->parentWidget();
+	QGroupBox *groupBoxCommands = parentWidget->findChild<QGroupBox*>("groupBoxCommands");
+	QGridLayout *commandGrid = groupBoxCommands->findChild<QGridLayout*>("commandGrid");
+	for (int i = 0; i < commandGrid->rowCount(); i++)
+	{
+		QWidget *w = NULL;
+		for (int j = 1; j < commandGrid->columnCount() && w == NULL; j++)
+		{
+			QLayoutItem *ql = commandGrid->itemAtPosition(i, j);
+			if (ql == NULL) continue;
+			QPushButton* pb = (QPushButton*)ql->widget();
+			if (pb == NULL) continue;
+			if (pb->property(PROPERTY_WIDGET_TYPE) == CG_PROGRAM) w = pb->property(PROPERTY_ASSOCIATED_INPUT).value<QWidget *>();
+		}
+		if (w == NULL) continue;
+		QString newCommandLine = tempCommands.value(getCmdID(w)).guessCommandLine();
+		if (!newCommandLine.isEmpty()) setText(w, newCommandLine);
+	}
+	buildManager->swapWinSearchPath(tmpMiktex, tmpTexlive);

```

Comment:
why is the path forgotten again ?

Reply:
To restore the original state before calling the function.